GitTank/GitProcessor.cs

```diff
@@ -45,126 +46,197 @@ public GitProcessor(IConfiguration configuration, ILogger logger)
             logger.Debug($"Repositories: {string.Join(", ", _repositories)}");
         }
 
-        ~GitProcessor()
+        private void OnOutput(int repositoryIndex, string line)
+        {
+            Output?.Invoke(repositoryIndex, line);
+        }
+
+        private ProcessHelper GetProcessHelper(int repositoryIndex, string workingDirectory)
+        {
+            var processHelper = new ProcessHelper(_logger, workingDirectory, repositoryIndex);
+            processHelper.Output += OnOutput;
+
+            return processHelper;
+        }
+
+        private void ReleaseProcessHelperUnmanagedResources(List<ProcessHelper> processHelpers)
         {
-            _processHelper.Output -= OnOutput;
+            foreach (var processHelper in processHelpers)
+            {
+                ReleaseProcessHelperUnmanagedResources(processHelper);
+            }
         }
 
-        private void OnOutput(string line)
+        private void ReleaseProcessHelperUnmanagedResources(ProcessHelper processHelper)
         {
-            Output?.Invoke(line);
+            processHelper.Output -= OnOutput;
         }
 
         public async Task<string> GetBranch()
```

Comment:
such methods could be renamed with async prefix
GetBranch -> GetBranchAsync
Update, Execute etc.